runtime/wasm/src/host_exports.rs

```diff
@@ -139,7 +141,20 @@ where
             entity_type,
             entity_id,
         };
-        ctx.state.entity_cache.set(key, Entity::from(data));
+
+        // Validate the changes against the subgraph schema
+        let updates = Entity::from(data);
+        let mut entity = ctx
+            .state
+            .entity_cache
+            .get(self.store.as_ref(), &key)
+            .map_err(|e| HostExportError(e.to_string()))?
+            .unwrap_or_else(|| Entity::new());
```

Comment:
What would you think of doing the `ctx.state.entity_cache.set(key, updates);` first, then you can just unwrap here, and the `entity.merge(updates.clone());` can be removed because it will be done in the entity cache.

Reply:
Good idea. If validation fails, the subgraph is toast anyway, and it doesn't matter what winds up in the `EntityCache`